texar/core/layers.py

```diff
@@ -647,6 +647,24 @@ def __init__(self, layers: Optional[List[nn.Module]] = None,
                 else:
                     self._layers.append(get_layer(hparams=layer))
 
+        self._functions = {
+            "sum": torch.sum,
+            "mean": torch.mean,
+            "prod": torch.prod,
+            "max": torch.max,
+            "min": torch.min,
+            "and": torch.all,
+            "or": torch.any,
+            "logsumexp": torch.logsumexp
+        }
+
+        def merge(tensors, dim):
+            _concat = torch.cat(tensors=tensors, dim=dim)
+            outputs = self._functions[self._mode](_concat, dim=dim)
+            return outputs
+
+        self._merge_op = merge
```

Comment:
I don't think it's necessary to store a function as attribute. The logic is simple and can be expanded in `forward` directly.

Reply:
Done!